### PR TITLE
[FLINK-35133][Connectors/Cassandra] Adding support for Flink 1.19

### PR DESCRIPTION
**Pull request by dannycranmer**

### CODE REVIEW

**Review by hlteoh37 — APPROVED** [review]
LGTM thanks for standardising the conectors @dannycranmer

File: .github/workflows/weekly.yml [hunk 1]
```diff
@@ -22,36 +22,24 @@ on:
     - cron: "0 0 * * 0"
   workflow_dispatch:
 jobs:
-  # tests that current connector iteration does not break compatibility with last 2 minor released Flink versions
-  non-main-version:
+  compile_and_test:
     if: github.repository_owner == 'apache'
+    strategy:
+      matrix:
+        flink_branches: [{
```
Comment (echauchot) [inline .github/workflows/weekly.yml:29]: It is true that all the PRs are tested against last 2 Flink versions but don't you think that the weekly workflow should also test against these versions and not only the snapshots ?

File: .github/workflows/weekly.yml [hunk 2]
```diff
@@ -22,36 +22,24 @@ on:
...
+          branch: main
+        }, {
+          flink: 1.20-SNAPSHOT,
+          branch: main
+        }]
     uses: apache/flink-connector-shared-utils/.github/workflows/ci.yml@ci_utils
     with:
-      flink_version: 1.17.2
-      connector_branch: main
-      skip_archunit_tests: true
```
Comment (echauchot) [inline .github/workflows/weekly.yml:32]: Cool that we no more need to test against 1.17.2 and skip the archunit tests !

**Review by echauchot — COMMENTED** [review]
reviewed after commit :cry: but LGTM. Just a single interrogation

File: .github/workflows/weekly.yml (hunk 1 above)
Comment (echauchot) [reply .github/workflows/weekly.yml:29]: perfect :+1:

**Pull request merged**